rust-engine/src/lean.rs

```diff
@@ -0,0 +1,442 @@
+//! The Lean backend
+//!
+//! This module defines the trait implementations to export the rust ast to
+//! Pretty::Doc type, which can in turn be exported to string (or, eventually,
+//! source maps).
+
+use crate::ast::span::Span;
+use crate::printer::Allocator;
+use std::iter::once;
+
+use pretty::{docs, DocAllocator, DocBuilder, Pretty};
+
+use crate::ast::identifiers::*;
+use crate::ast::literals::*;
+use crate::ast::*;
+
+macro_rules! print_todo {
+    ($allocator:ident) => {
+        $allocator.text(format!("Todo at {}", line!())).parens()
+    };
+}
+
+const INDENT: isize = 2;
+
+/// Placeholder structure for lean printer
+pub struct Lean;
+
+/// Implementations for each part of the AST
```

Comment:
Doc strings are always attached to items: they are never floating.
If you want to group impls, you have to use a `mod` or a `const _: () = {};`.

But frankly, not sure this would be useful here: a printer is really all about implementing the `Pretty` trait again and again. So dropping the comment seems fine.